src/Symfony/Component/Mailer/Envelope.php

```diff
@@ -85,4 +81,29 @@ public function getRecipients(): array
     {
         return $this->recipients;
     }
+
+    /**
+      * Returns true if any address' localpart contains at least one
+      * non-ASCII character, and false if all addresses have all-ASCII
+      * localparts.
+      *
+      * This helps to decide whether to the SMTPUTF8 extensions (RFC
+      * 6530 and following) for any given message.
+      *
+      * The SMTPUTF8 extension is strictly required if any address
+      * contains a non-ASCII character in its localpart. If non-ASCII
+      * is only used in domains (e.g. horst@freiherr-von-mühlhausen.de)
+      * then it is possible to to send the message using IDN encoding
+      * instead of SMTPUTF8. The most common software will display the
+      * message as intended.
+      */
+    public function anyAddressHasUnicodeLocalpart(): bool
+    {
+        if($this->getSender()->hasUnicodeLocalpart())
+            return true;
+        foreach($this->getRecipients() as $r)
+            if($r->hasUnicodeLocalpart())
```

Comment:
using this method requires updating the min requirement of `symfony/mime` in the `mailer` component (requiring `^7.2`). This is why the low-deps CI job is failing right now.

Reply:
Fixed, thanks.